src/plugins/vis_default_editor/public/agg_filters/agg_type_field_filters.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IAggConfig, IndexPatternField } from '../../../data/public';
+
+type AggTypeFieldFilter = (field: IndexPatternField, aggConfig: IAggConfig) => boolean;
+
+const filters: AggTypeFieldFilter[] = [
+  /**
+   * If rollup index pattern, check its capabilities
+   * and limit available fields for a given aggType based on that.
+   */
```

Comment:
The only other plugin registering to the agg type/field filters registries was `rollup`, and because the `aggTypeFilters` registry was accidentally exported statically (instead of on the runtime contract), it is possible these filters were not all even being applied in the editor as they could have been included in separate bundles. 

It feels a little weird to add the filtering logic for these here (in OSS), however as we have discussed previously, I don't think filters justify their own registry, when it is essentially a concern of the editor UI, and the editor is the only consumer of the registry.

One alternative I could see here would be to convert it back to a registry exposed from vis_default_editor.

Reply:
I think the current implementation is fine